### ISSUE TITLE
non-primitive cast: `std::num::NonZeroU64` as `u32` error

### ISSUE THREAD

**Issue by russcam** [rust-lang/rust-clippy#5388]
- rustc 1.44.0-nightly (77621317d 2020-03-28)
- cargo 1.44.0-nightly (8a0d4d9c9 2020-03-24)
- clippy 0.0.212 (70b93aa 2020-03-27)

Clippy is emitting the following error when run as part of a GitHub clippy action:

```
error[E0605]: non-primitive cast: `std::num::NonZeroU64` as `u32`
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_data_structures-642.0.0/profiling.rs:471:25
    |
471 |         let thread_id = std::thread::current().id().as_u64() as u32;
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: an `as` expression can only be used to convert between primitive types. Consider using the `From` trait
```

More details can be found in [the GitHub action result](https://github.com/elastic/elasticsearch-rs/runs/543929263?check_suite_focus=true)

**Comment by flip1995** [comment]
That is a compiler error, not a Clippy error. That being said, it seems to trigger on  a dependency, which shouldn't happen at all 🤔

**Comment by flip1995** [comment]
Marking with help wanted and needs test, since I'm not sure if it is a Clippy issue, a action-rs/clippy-check issue or a cargo/rustc issue

**Comment by pmk21** [comment]
Came across this error while trying to install `racer`.

I believe there seems to be a bug in the `rustc-ap-rustc_data_structures` crate. In the version `642.0.0`.

Did some research and found that a `.get()` seems to be missing in the statement `std::thread::current().id().as_u64()` which throws up the error.

Seems to have been fixed in version `649.0.0` of the crate.

**Comment by flip1995** [comment]
Thanks! so this is not a Clippy bug. 🎉
